MovieQuiz/Presentation/MovieQuizViewController.swift

```diff
@@ -3,27 +3,85 @@ import UIKit
 final class MovieQuizViewController: UIViewController {
     private var state = GameState()
 
+    private var questionFactory: QuestionFactoryProtocol?
+    private var resultPresenter: ResultPresenterProtocol?
+    private var statisticService: StatisticsService?
+
     @IBOutlet private weak var imageView: UIImageView!
     @IBOutlet private weak var textLabel: UILabel!
     @IBOutlet private weak var counterLabel: UILabel!
 
     @IBOutlet private weak var yesButton: UIButton!
     @IBOutlet private weak var noButton: UIButton!
 
-    @IBOutlet private weak var dimView: UIView!
+    private var activityIndicator = {
+        let indicator = UIActivityIndicatorView()
+        indicator.isHidden = true
+
+        return indicator
+    }()
```

Comment:
У вас работает этот код?

Пожалуйста, добавьте явно тип. Компилятору сложно определить из нескольких строк что же за тип нам вернется. 
```suggestion
    private var activityIndicator: UIActivityIndicatorView = {
        let indicator = UIActivityIndicatorView()
        indicator.isHidden = true

        return indicator
    }()
```

Вторая вещь, isHidden хорошо решает проблему скрытия, когда индикатор не нужен. Однако писать такой код каждый раз было бы лень, и Apple подарила нам hidesWhenStopped, которое делает то же самое одной строчкой.

Reply:
@ABashkirova по поводу неявного типа: я сам и не думал, что оно сработает, но у меня оно сработало, видать дело в свежей комбинации xcode/swift, хотя вроде как настроил что проект должен открываться и в старых икскодах. В любом случае исправил на надежный классический вариант :)

Reply:
@ABashkirova спасибо за наводку про `hidesWhenStopped`, это все сильно упрощает :)